### DISCUSSION TITLE
Example of creating a Pyside package for windows

### DISCUSSION THREAD

**Discussion by araifractal** [AcademySoftwareFoundation/rez#1308] [Q&A]
Can We have an example of how to create a package for PySide2 for windows? It would give a proper perspective of how to approach building packages of this nature.

**Comment by hughetop** [comment]
I believe you can just do `rez-pip --install PySide2` and it will use whatever version of python rez uses. Alternatively, you can specify the python version with `--python-version VERSION`. Test to make sure it works, then you can do `rez-pip --release PySide2`.

**Reply by ColinKennedy** [reply]
FWIW `rez-pip --install PySide2 --python-version=3 --release` doesn't work for me.

```
Windows PowerShell
Copyright (C) Microsoft Corporation. All rights reserved.

Try the new cross-platform PowerShell https://aka.ms/pscore6

PS C:\Users\korinkite> rez-pip --install PySide2 --python-version=3 --release
13:23:06 INFO     Trying to use pip from python package
13:23:06 INFO     Trying to use pip from pip package
13:23:06 WARNING  Found no pip in any python and/or pip rez packages!
13:23:06 WARNING  Falling back to pip installed in rez own virtualenv:
13:23:06 WARNING         pip: 19.1.1 (c:\users\korinkite\rez-2.104.7-python-2.7\lib\site-packages\pip\__init__.pyc)
13:23:06 WARNING      python: 2.7.16 (c:\users\korinkite\rez-2.104.7-python-2.7\scripts\python.exe)
13:23:06 INFO     Installing 'PySide2' with pip taken from 'c:\\users\\korinkite\\rez-2.104.7-python-2.7\\scripts\\python.exe'
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.
Collecting PySide2
  ERROR: Could not find a version that satisfies the requirement PySide2 (from versions: none)
ERROR: No matching distribution found for PySide2
WARNING: You are using pip version 19.1.1, however version 20.3.4 is available.
You should consider upgrading via the 'python -m pip install --upgrade pip' command.
13:23:07 ERROR    BuildError: Failed to download source with pip: 'c:\users\korinkite\rez-2.104.7-python-2.7\scripts\python.exe' -m pip install --use-pep517 '--target=c:\users\korink~1\appdata\local\temp\pip-n1ob_c-rez' PySide2
PS C:\Users\korinkite>
```

It didn't work out of box for me, even though installing other packages like `Sphinx` works just fine. I can see in the log that it's falling back from Python version 3 to version 2 (from Rez's own python). IMO this is bad design, I asked for Python 3 and it should use Python 3. But either way, a guide on troubleshooting rez-pip, in general, I think is needed. The current errors for rez-pip could be more prescriptive (recommend to the user what to do to fix the problem) so more people can use rez-pip more effectively.

**Reply by hughetop** [reply]
Looks like the python version syntax is different, I did `rez-pip --install PySide2 --python-version 3.9` and it installed PySide2 and shiboken2 to my home directory, with dependencies on python-3.9. Note that the python version you specify must be an available rez package.

**Reply by ColinKennedy** [reply]
You're describing the expected behavior, yes. The tool itself is rolling down python versions even after I'd specified the major version to use. This and other things negatively affect UX so I agree with @araifractal that examples / tutorials / docs are definitely needed. `rez-pip` is one of the first tools people use when they set up Rez at a company, it should be easy to use, give meaningful feedback during unexpected failure, and be well docced - is all I'm advocating for.
